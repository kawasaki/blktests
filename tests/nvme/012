#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# Test mkfs with data verification for block device backed ns.

. tests/nvme/rc
. common/xfs

DESCRIPTION="run mkfs and data verification fio job on NVMeOF block device-backed ns"
TIMED=1

requires() {
	_nvme_requires
	_have_xfs
	_have_fio
	_have_loop
	_require_nvme_trtype_is_fabrics
	_require_nvme_test_img_size 350m
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local nvmedev

	_nvmet_target_setup

	_nvme_connect_subsys "${nvme_trtype}" "${def_subsysnqn}"

	nvmedev=$(_find_nvme_dev "${def_subsysnqn}")
	_check_uuid "${nvmedev}"

	_xfs_run_fio_verify_io "/dev/${nvmedev}n1"

	_nvme_disconnect_subsys "${def_subsysnqn}"

	_nvmet_target_cleanup

	echo "Test complete"
}
