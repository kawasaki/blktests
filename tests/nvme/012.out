Running nvme/012
disconnected 1 controller(s)
Test complete
